spec/controllers/saml_idp_controller_spec.rb

```diff
@@ -203,23 +197,22 @@
     end
 
     describe 'HEAD /api/saml/auth', type: :request do
-      it 'responds with "400 Forbidden" with unknown bindings' do
-        head '/api/saml/auth?SAMLRequest=bang!'
-
-        expect(response.status).to eq(403)
+      it 'raises an invalid service provider error' do
+        expect { head '/api/saml/auth?SAMLRequest=bang!' }.
+          to raise_error 'nil is an invalid service provider!'
```

Comment:
Shouldn't we handle this and return an http error?  My understanding is that when assertions show up at runtime like this that points to bug in the code (at least thats how think about it in python world).